tests/functional/adapter/test_ephemeral.py

```diff
@@ -0,0 +1,10 @@
+import pytest
+from dbt.tests.adapter.ephemeral.test_ephemeral import BaseEphemeralMulti
+from dbt.tests.util import run_dbt, check_relations_equal
+
+class TestEphemeralMultiSnowflake(BaseEphemeralMulti):
+
+    def test_ephemeral_multi_snowflake(self, project):
+        results = run_dbt(["run"])
+        assert len(results) == 3
+        check_relations_equal(project.adapter, ["SEED", "DEPENDENT", "DOUBLE_DEPENDENT", "SUPER_DEPENDENT"])
```

Comment:
Biggest difference between the `Multi` test in` core `and here is the way the `relation` name gets uppercased

Reply:
I'm assuming snowflake needs these uppercased?  Are there other tests that do something similar?  How do they handle it?

Reply:
`test_warehouses` has a similar part of the test here https://github.com/dbt-labs/dbt-snowflake/blob/main/tests/functional/warehouse_test/test_warehouses.py#L80-L85 looks like the models being lowercase initially shouldn't be a issue because it compares after they have been run which would return them upper

Reply:
Or you can make the things to check(`["SEED", "DEPENDENT", "DOUBLE_DEPENDENT", "SUPER_DEPENDENT"]`) a fixture, then overwrite it in dbt-snowflake